Src/FluentAssertions.Json/JSonAssertionOption.cs

```diff
@@ -0,0 +1,8 @@
+﻿using FluentAssertions.Equivalency;
+
+namespace FluentAssertions.Json
+{
+    public sealed class JSonAssertionOption<T> : EquivalencyAssertionOptions<T> , IJsonAssertionOptions<T>
```

Comment:
🤔 On second thought, this isn't going to work either. As soon as you call `WhenTypeIs`, you'll return the original `EquivalencyAssertionOptions`. I think it's better to completely hide the existence of the FA internals. So, your `IJsonAssertionOptions<T>.Using` should return a wrapper class that uses the `EquivalencyAssertionOptions<T>.Restriction<TProperty>` class internally.